Создал файл test_cases_button_demoqa.txt

IlyaKirdun opened the pull request:
Добавил button.page.ts , button.spec.ts (4 cases).

4 для страницы Button:

1. Проверка функционала кнопки "Double Click Me".
2. Проверка функционала кнопки "Right Click Me".
3. Проверка функционала кнопки "Click Me".
4. Проверка сохранения отображения сообщения при нажатии кнопок: "Right Click Me", "Click Me", "Double Click Me".

IlyaKirdun committed:
Создал файл test_cases_button_demoqa.txt

rybignat commented on testCases/test_cases_button_demoqa.txt:
```diff
@@ -0,0 +1,58 @@
+
+---
+
+Test Case 3: Проверка функционала кнопки "Click Me".
+Preconditions:
+- Открыть браузер и перейти на 'https://demoqa.com/'.
+- Перейти в "Elements", затем нажать на "Button" в меню навигации.
+
+Steps:
+1. Нажимаем левой кнопкой по кнопке "Click Me".
+Expected result:
+- Кнопка нажата.
+
+2. Проверяем наличие сообщения "You have done a dynamic click".
+Expected result:
+- Сообщение отображается.
+
+Post-condition:
+- Закрыть браузер.
+
+---
```
Сообщение о "dynamic click" не навело ни на какие мысли?

IlyaKirdun committed:
Переработал тест "Test Case 4: Проверка сохранения отображения сообщения при нажатии кнопок: "Right Click Me" и "Click Me".",
Переработал тест "Test Case 3: Проверка функционала кнопки "Click Me"."

IlyaKirdun commented:
Учёл момент с динамической кнопкой, жду реквест)

rybignat reviewed (changes requested):
Ну,4 кейс должен проверять как себя ведет поле инпута, запоминает оно клики, не запоминает...
А этот кейс должен проверятся в рамках 3-го

IlyaKirdun committed:
Удалил example.spec.ts,
Добавил метод reloadPage() в functions.ts
Создал файл button.page.ts

rybignat commented on page/button.page.ts:
```diff
@@ -0,0 +1,34 @@
+        await this.rightClickButton.click({ button: "right"})
+    }
+
+    async checkVisibilityButtonMessage(buttonName: 'double' | 'right' | 'dynamic', state: VisibilityState): Promise<void> {
+        const buttonMessage: Locator = this.page.locator(`//p[@id="${buttonName}ClickMessage"]`)
+
+        await expect(buttonMessage)[state]()
+    }
```
Круто, а как ты узнаешь что у тебя сообщение корректное? Может там пьяный фронт текст поменяет на "Илья, твои тесты - не работают". А ты и не узнаешь)
Так же не вижу обновления версий в package.json

IlyaKirdun committed:
Импортировал метод assertByState(),
Переработал метод verifyMatchButtonMessage()

IlyaKirdun committed:
Создал файл button.spec.ts
Удалил неактуальные методы в functions.ts
Поправил логику метода verifyMatchButtonMessage() в button.page.ts

IlyaKirdun committed:
Переработал метод getCurrentDynamicButtonId()
Исправил опечатки
Изменил локатор rightClickButton
Переработал тест "Проверка сохранения отображения сообщения при нажатии кнопок: "Right Click Me", "Click Me", "Double Click Me"."
Обновил версию в package.json
